graphql_client/__init__.py

```diff
@@ -30,7 +30,8 @@
 GQL_COMPLETE = 'complete'
 GQL_CONNECTION_KEEP_ALIVE = 'ka'
 
-logging.basicConfig()
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
```

Comment:
Can you please elaborate a bit about the motivation for this change? 

From the first paragraph, of the docs that you linked (quoting the important parts for brevity):

> If the using application does not use logging, and library code makes logging calls...will be printed to sys.stderr. This is regarded as the best default behaviour.


Then it goes on to say: 
> If for some reason you don’t want these messages printed .... you can attach a do-nothing handler

So I am assuming, you want the default behaviour of this library to not print anything. May I know why? Because the docs seem to say the former is the best default behaviour.

I don't really have any strong opinions. I am just trying to understand your motivation/use-case.

Reply:
Thanks for question!
OK, look, two points here: **basicConfig** and **NullHandler**.

as for **basicConfig**:
`loggin.basicConfig()` creates a root logger with default parameters: a format, a level, a handler. Usually application code does it, not libraries/dependencies, because it knows better how to configure it. Better explanation from the docs:

> Note It is strongly advised that you do not add any handlers other than NullHandler to your library’s loggers. This is because the configuration of handlers is the prerogative of the application developer who uses your library. The application developer knows their target audience and what handlers are most appropriate for their application: if you add handlers ‘under the hood’, you might well interfere with their ability to carry out unit tests and deliver logs which suit their requirements.

Technically it's possible change a root logger (https://stackoverflow.com/questions/12158048/changing-loggings-basicconfig-which-is-already-set), but why?
As an example you may look at `websocket` package, which is a dependency of `graphql-client`
https://github.com/websocket-client/websocket-client/blob/master/websocket/_logging.py


as for **NullHandler**:
It didn't print any log messages the absence of any logging configuration. I.e. if i don't configured logging in my app, i won't get logs from the library.
If I do a configuration in my app by 'basicConfig()', I'll get them according to levels, because my logger will have a real handler, and this handler will print messages; that's because of logger hierarchy and propagation. 
So I don't want to 'no printing anything'. I want to leave this question to using app.

Reply:
Sorry for the delay! I'll get back to you ASAP

Reply:
man, just merge it and that's all :wink: